### ISSUE TITLE
Interrupted page loads and ERR_NETWORK_CHANGED

### ISSUE THREAD

**Issue by aaronovz1** [brave/brave-browser#5958]
## Description 
Since the last few releases I get ERR_NETWORK_CHANGED errors when loading all kinds of websites. I also see many pages only partly load. I have to reload the page a couple times to get passed this issue.

JIRA Cloud by Atlassian is probably the most problematic with random drop outs during page load or form submissions. I suspect it is because it has quite a complex UI and lots happening in the background.

I have seen this on two versions of Linux and also on the Windows build. All running the same version of Brave.

## Steps to Reproduce
No exact steps to reproduce. Happens randomly on any website.

## Actual result:
Your connection was interrupted
A network change was detected.
ERR_NETWORK_CHANGED

## Expected result:
Pages load completely without interruption.

## Reproduces how often: 
Multiple times a day.

## Brave version (brave://version info)

Brave | 0.68.132 Chromium: 76.0.3809.132 (Official Build) (64-bit)
-- | --
Revision | fd1acc410994a7a68ac25bc77513d443f3130860-refs/branch-heads/3809@{#1035}
OS | Linux
JavaScript | V8 7.6.303.29
Flash | (Disabled)
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? N/A
- Can you reproduce this issue with the dev channel? N/A
- Can you reproduce this issue with the nightly channel? N/A

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? Don't think so
- Does the issue resolve itself when disabling Brave Rewards? Not sure
- Is the issue reproducible on the latest version of Chrome? Not sure

**Comment by aaronovz1** [comment]
Possibly an issue IPv6

https://superuser.com/questions/747735/regularly-getting-err-network-changed-errors-in-chrome

I'm going to try disabling IPv6 as suggested and see if the problem goes away. Did anything change in Brave surround IPv6?

**Comment by rebron** [comment]
@aaronovz1 did disabling IPv6 help? problem go away?

**Comment by aaronovz1** [comment]
Yes, it did seem to workaround the problem.

**Comment by bsclifton** [comment]
@aaronovz1 awesome, thanks for the update! 😄

**Comment by aaronovz1** [comment]
Is that really the solution though? Why is Brave affected by IPv6 being enabled? It was also only affected in recent versions. I don't think Firefox and other browsers had this issue.

**Comment by bsclifton** [comment]
@aaronovz1 good call- will update issue with better description. I'm not sure why it's affected

**Comment by jamesdlow** [comment]
Just reporting that I'm having an issue with this on a Mac. Only happens in Chrome and Brave, Safari and Firefox are fine. Resetting my router does help for a while, but then the issue comes back. Disabling IPv6 seemed to be a work around.

**Comment by aaronovz1** [comment]
FYI - I am still seeing this on the latest release version.

**Comment by danielcgold** [comment]
+1 to still seeing this issue

**Comment by betoval** [comment]
i have the same issue on version 1.18.77

**Comment by sup2069** [comment]
Getting the same issue on V1.20.103.

**Comment by danielcgold** [comment]
For me, this didn't turn out to be an issue with `IPv6`.

I couldn't track down the exact instance that cause my error, but I updated Homebrew and all of my formulas and the problem went away. I ruled out this being caused by any hardware, OS and browser version issues by creating a new profile on my Mac and the `ERR_NETWORK_CHANGED` problem didn't exist.

Hope this is helpful for someone else.

**Comment by psycry** [comment]
Still getting this, it seems to coincide with highly tracked sites (bbc.com won't load, cnn, fox, all the news heavy subsidized ad revenue news outlets).  If you turn off the blockers it works like chrome but I'm pretty sure it's something about the way some of their ads/trackers redirect traffic and brave blows up when it detects it.

Github doesn't load worth a $#@% either just FYI but it's always done that, especially behind VPNs.

**Comment by psycry** [comment]
Another side note, if you disable the brave blockers, load bbc.com (loads fine), and then re-load it with blockers back on, it loads.  Maybe there's a cookie or something it needs set before it clears the traffic?

**Comment by naezeroth** [comment]
I am still getting this issue too, it affects all devices in my network - only Brave is seeing this issue whereas safari or other browsers are not. I'm also not super sure what causes it I have a Pihole running too, but will test with ipv6 off.

**Comment by wazum** [comment]
I can still confirm this occasionally with Chrome 1.31.91 on Mac OS 10.15,
disabling IPv6 seems to resolve the problem.

**Comment by axeloz** [comment]
Hello, I started using Brave 5 days ago. Everything worked, until this morning. Many many many ERR_NETWORK_CHANGED errors. Safari and FF are working just fine. I restarted the router, and my Mac but no luck. Thanks

**Comment by axeloz** [comment]
Hello, I started using Brave 5 days ago. Everything worked, until this morning. Many many many ERR_NETWORK_CHANGED errors. Safari and FF are working just fine. I restarted the router, and my Mac but no luck. Thanks

**Comment by agrawalsourav98** [comment]
Been facing this issue for some time now.

**Comment by benrudhart** [comment]
Unfortunately I'm also running into this issue lately. Do we have any recommendations for working workarounds?

**Comment by agrawalsourav98** [comment]
Only solution being is to disable IPv6.

**Comment by mattheworiordan** [comment]
I have had this issue for weeks now. 

I get the following loading some pages:
![MO screenshot 2023-01-30 at 16 15 15](https://user-images.githubusercontent.com/43789/215571090-34c57fb6-c7c8-4790-bfa1-df3a1011a124.png)

But looking in the dev console, I can see request randomly fail due to this, and I get failed assets/XHR requests etc. Reloading normally works for a short while.

![MO screenshot 2023-01-30 at 16 16 09](https://user-images.githubusercontent.com/43789/215571239-b4ac4292-cc76-406d-8fef-a44f23366144.png)

I can reproduce this easily on Chrome and Brave, but not on Safari.  On a Mac too. Very annoying, will try the IPv6 disabling.

**Comment by benrudhart** [comment]
One workaround that worked for me: In case you're connecting your mac to internet via two ways (e.g. wifi + ethernet) -> disable one of them, e.g. wifi.

**Comment by auzystephens** [comment]
I am also experiencing this issue on MacOS Ventura 13.4. Issue also limited to Brave, other browsers seem fine. Testing the IPV6 trick but definitely not a long term solution.

**Comment by 4ydx** [comment]
I see this as well and will try the disable ipv6 suggestions.  This bug is annoying.

**Comment by 4ydx** [comment]
I see this as well and will try the disable ipv6 suggestions.

**Comment by steakknife** [comment]
Yep. Brave doesn't support IPv6 properly in 2023. Pretty embarrassing and unusual when more and more large environments are going IPv6 single-stack due to scale issues.

**Comment by adamJLev** [comment]
This issue is driving me absolutely insane, at this point I guess i'll just switch to Safari after years of enjoying Brave as my main browser.

Requests will randomly fail , and it only happens in Brave- not Safari, not with Terminal commands, etc

I've tried disabling ipv6 but that didnt fix it either: https://protonvpn.com/support/how-to-disable-ipv6-protocol-on-macos/

To try to narrow down the issue, I run this command in my terminal in the background and even when Brave shows me this error- there's no issues in my `curl` command that executes every 0.1 seconds:
```
while true; do curl  --silent --output /dev/null --show-error --fail --connect-timeout 0.5 https://www.google.com; sleep 0.1; done
```

Clearly something is wrong in Brave or its chromium upstream network code.

**Comment by skull-squadron** [comment]
Disabling IPv6 on the L2 and L3 network and disable IPv6 name resolution
transport and AAAA resolution at the router might resolve the issue. (Work
has infrastructure subnets that are IPv6 only so I guess I'm spoiled.)

Also not IPv6 ready:
- Github Copilot
- Amazon Echo Show 15

There's always Firefox. Safari uses an old custom Apple cryptographic
library.

**Comment by houmark** [comment]
@adamJLev This may be a completely different issue and not related to IPv6. I had this issue with both Chrome and Brave but not in Safari and I also went through disabling everything related to IPv6 with no improvement.

I had connected my iPhone through USB to my Mac and that resulted in a very weird process starting and running in the background, which would rewrite settings related to my primary internet connection, making both Chrome and Brave throw this error very often.

In a Terminal try running this command and let it hang there and monitor it. When the error happens you may see a changed key in Network/Global/IPv4:
`(echo -e "n.add State:/Network/Global/IPv4\nn.watch" & cat) | scutil`

It's been a few months since I had this issue and there's no permanent solution or fix that I know of, only killing the process and avoiding connecting your phone works. I am keeping this command running in my terminal at all times to monitor this ongoing (it's normal that it will output a line or two when you change wifi or adaptor, going from wifi to ethernet).

Let me know if you are seeing lines output often (no output means no changes to your internet configuration). If you do, then I can look back in my notes and guide you on what I did to kill the rouge process.

**Comment by abayomi185** [comment]
I'm also facing this issue, making Brave unusable for me. It seems like it might be related to this upstream issue:
https://bugs.chromium.org/p/chromium/issues/detail?id=1498910#c19

**Comment by adamJLev** [comment]
Yes it's definetely iPhone + USB related, my workaround for now has been to either not connect my iPhone to my laptop - or if I need to charge it, when prompted on the phone with "do you trust this computer " I tap on "don't trust".
Annoying but less annoying than the other issue

**Comment by avandenberghe** [comment]
This issue still happens every day. 
[Version 1.61.120 Chromium: 120.0.6099.234  (arm64)](https://brave.com/latest/)
<img width="720" alt="Scherm­afbeelding 2024-01-23 om 13 32 49" src="https://github.com/brave/brave-browser/assets/226730/171e649d-a18f-45c3-aba8-cfe85b39419d">

**Comment by cavricks** [comment]
Same here, I started to think it was network for a while, maybe DNS settings. Until it started to happen on a different pc of mine in a different network and ISP. Safari works fine, everything works. Brave randomly throwing ERR_NETWORK_CHANGED errors. It has been driving me crazy.

**Comment by DPangerl** [comment]
For me also downloads are interrupted like every 10 sec.
Doesn't depend on any equipment attached to my MacBook.

**Comment by petrikaj** [comment]
This has become unbearable. Almost every page load brings up this error.

**Comment by rvanlaarhoven** [comment]
> For me, this didn't turn out to be an issue with `IPv6`.
> 
> I couldn't track down the exact instance that cause my error, but I updated Homebrew and all of my formulas and the problem went away. I ruled out this being caused by any hardware, OS and browser version issues by creating a new profile on my Mac and the `ERR_NETWORK_CHANGED` problem didn't exist.
> 
> Hope this is helpful for someone else.

I think this is what solved it for me too. I ran a `brew upgrade` on some of the outdated dependencies (see `brew outdated`) and haven't experienced the issue since.

**Comment by mk-conn** [comment]
<img width="1719" alt="Screenshot 2024-03-13 at 13 29 14" src="https://github.com/brave/brave-browser/assets/5840922/6226f528-0aec-4866-94d2-541b3b907077">

Is this being worked on? Brave is barely usable anymore...

**Comment by itspaulie** [comment]
Yes, it has been very annoying for quite some time. Looks like chromium bug [#40939844](https://issues.chromium.org/issues/40939844).

**Comment by skull-squadron** [comment]
Not seeing it lately. Possible contributing factors:

0.  LuLu by Objective-See was a source of nondeterministic packet loss
affecting ssh and more. This has been removed.

1. Dual stack ISP and Wi-Fi. Had to downshift device to IP4-only for
compatibility with a particular development tool.

2. Ethernet and Wi-Fi active while docked to monitor.

**Comment by skull-squadron** [comment]
I spoke too soon. Happening a lot today. Changing browsers ensues.

**Comment by adamJLev** [comment]
That sounds about right, any updates from Brave team on this?
Its driving me crazy, I even updated my mac to Sequoia hoping this would fix it, but its still happening

**Comment by lrosset** [comment]
it is indeed now a nightmare and impossible to use some essential webapp like OpenAI with Brave + OSX Sequoia.  I switched to Firefox Dev Edition to get these working until Brave dares to fix this like Google Chrome did yet.

https://community.brave.com/t/macos-sequoia-needs-to-import-googles-fix-for-err-network-changed/569151

**Comment by bsclifton** [comment]
Checking this out - just a heads up though, the `ReduceIPAddressChangeNotification` feature introduced with https://chromium-review.googlesource.com/c/chromium/src/+/5348265 has been enabled by default since mid-March 2024:
https://source.chromium.org/chromium/chromium/src/+/97338d7dea7a930ef892c95d0ff73d9c89958c7e

Is this working on Chromium browsers for folks? But not on Brave? Or problems there too - but maybe it happens in Brave more often? I am inspecting some of the network patches we have to see if we deviate in any way

**Comment by lrosset** [comment]
That's really weird indeed. So issues at all since too although I didn't updated Brave at all nor did I update Sequoia OSX to its last version today. I just disabled the firewall OSX in order to stop all interferences with Little Snitch. My broadband provider didn't noticed any issues on it infrastructure, and yet, I am still wondering if this wasn't something else that has been fixed somewhere else... Weird

**Comment by adamJLev** [comment]
disabling the macos Firewall (in network settings) did seem to work last week when I was still having the issue.
but now it works even with macos Firewall enabled

**Comment by AndreaDiazCorreia** [comment]
I have this problem in brave with linux ubuntu 24.04, it is not solved yet?

**Comment by bsclifton** [comment]
Good to know the macOS case should be solved 😄 

Definitely seems like there is still a problem on Linux. @AndreaDiazCorreia as a temporary work-around, you can disable IPv6 if you are not using it. But seems there is still an underlying root cause.

**Comment by absolutegravitas** [comment]
still an issue on arch linux (endeavouros) running Brave on KDE -- makes browser unusable

**Comment by cfharris** [comment]
I'm having this issue on virtually every page open or refresh. This is Brave 1.73.91 which says is current on Linux Mint 21.3 (XFCE). I also have this same problem in Chrome as well, so it definitely seems like the root of the issue is there.

**Comment by angelomelonas** [comment]
Like @cfharris , I am also seeing this almost on every page open or refresh. Running Brave Version 1.73.105 Chromium: 131.0.6778.265 (Official Build) (64-bit) and Ubuntu 24.04.1 LTS.

**Comment by khatriafaz** [comment]
I was also having this issue almost everyday. I have disabled IPv6 on OS level (I am on Ubuntu 24.04) after which it is working properly for now.

**Comment by mcgroarty** [comment]
@khatriafaz It's a shot in the dark. But any chance you're using an Eero router? If so, try changing:

Network Settings → Thread → Off

This creates a storm of IPv6 route announcements, which causes Chrome on Linux and macOS to see a network topology changes. (Windows filters these events unless the primary route actually changes as a result of the announcements, so it hasn't been similarly affected.)

You can watch these on Linux with ip -6 route show | ts

I suspect the fact that this traffic makes it through to the non-Thread interfaces on the Eero is a defect.

**Comment by Eunanibus** [comment]
I'm also getting this. I've also narrowed this down to a Brave issue. It makes experiences like completing form submissions or loading one-time links utterly infuriating. This should be a higher priority. I'm honestly thinking this might push me to an alternative browser.

**Comment by micheljung** [comment]
In my case, this was caused by a Docker container being in a restart loop, which resulted in network interfaces being created and removed continuously. This combined with https://github.com/brave/brave-browser/issues/5958#issuecomment-2392515133 seemed to cause the issue.
